Data/ByteString/Lazy.hs

```diff
@@ -609,11 +620,43 @@ scanl
     -- ^ input of length n
     -> ByteString
     -- ^ output of length n+1
-scanl f z = snd . foldl k (z,singleton z)
- where
-    k (c,acc) a = let n = f c a in (n, acc `snoc` n)
+scanl function = fmap (uncurry (flip snoc)) . mapAccumLChunks (S.mapAccumL (\x y -> (function x y, x)))
 {-# INLINE scanl #-}
 
+-- | 'scanl1' is a variant of 'scanl' that has no starting value argument.
+-- This function will fuse.
```

Comment:
What is this claim based on? Is there a test?

Reply:
I think the claim about fusion is false. This is tracked in #374. I imagine I must have copied this comment from somewhere without giving it a thought, I am going to fix it.